Runtime/ApiClientConnection.cs

```diff
@@ -567,11 +567,44 @@ public HttpClientByteArrayRequest CreateGetByteArrayRequest(
                 _urlCache,
                 cachePolicy,
                 () => this.CreateGetByteArrayRequest(
-                    url, 
-                    ct, 
-                    authentication, 
+                    url,
+                    ct,
+                    authentication,
+                    headers,
+                    useDefaultHeaders,
+                    cachePolicy))
+            {
+                Authentication = authentication,
+                Headers = headers,
+                DefaultHeaders = useDefaultHeaders ? _defaultHeaders : null
+            };
+
+            return request;
+        }
+
+        public HttpClientHeadersRequest CreateGetHeadersOnlyRequest(
+            string url,
+            CancellationToken ct,
+            AuthenticationHeaderValue authentication = null,
+            Dictionary<string, string> headers = null,
+            bool useDefaultHeaders = true,
+            CachePolicy cachePolicy = null)
+        {
+            var request = new HttpClientHeadersRequest(
+                new HttpRequestMessage(HttpMethod.Get, url)
```

Comment:
instead of sending GET request you should be able to send HEAD request to get the headers without the response body.
When you send HEAD request the server should not even send the body of the response over the network and this should make HEAD request faster and using less data while still providing response headers.